be/src/olap/segment_loader.cpp

```diff
@@ -37,26 +37,22 @@ bool SegmentCache::lookup(const SegmentCache::CacheKey& key, SegmentCacheHandle*
     if (lru_handle == nullptr) {
         return false;
     }
-    handle->init(_cache.get(), lru_handle);
+    handle->push_segment(_cache.get(), lru_handle);
```

Comment:
We need a lock in push_segment?

Reply:
Not need. The concurrency occurs only at the methods of LRUCache. The variable `handle` is only used by one thread.